tests/Integration/Queue/UniqueJobTest.php

```diff
@@ -48,9 +48,9 @@ public function testUniqueJobsAreNotDispatched()
             $this->app->get(Cache::class)->lock($this->getLockKey(UniqueTestJob::class), 10)->get()
         );
 
-        Bus::fake();
+        Bus::assertDispatchedTimes(UniqueTestJob::class);
         UniqueTestJob::dispatch();
-        Bus::assertNotDispatched(UniqueTestJob::class);
+        Bus::assertDispatchedTimes(UniqueTestJob::class);
```

Comment:
Why was this test changed?

Reply:
@taylorotwell because of the changes made here:

```php
// \Illuminate\Support\Facades\Bus.php

return tap(new BusFake(static::getFacadeRoot(), $jobsToFake, $batchRepository), function ($fake) {
    if (! static::isFake()) {
        static::swap($fake);
    }
});
```

Previously, this test:
1. Swapped the underlying facade root instance for `BusFake` with a `$dispatcher` property of the previous facade root (`\Illuminate\Bus\Dispatcher`) – Line 42
3. Swapped out the faked facade root for a new `BusFake` instance, now with a `$dispatcher` property of the previous `BusFake` instance from the facade root – Line 51.

As a result of the fix, calling `Bus::fake()` twice in the same test now causes `Bus::assertNotDispatched(...)` to fail, which is correct, as we're now using the same `\Illuminate\Contracts\Bus\QueueingDispatcher` instance for the `BusFake::$dispatcher` property.

This test is now correct because, despite executing `UniqueTestJob::dispatch();` twice (lines 44 and 52), we are now correctly verifying that the unique job was dispatched exactly once on the underlying `BusFake` facade instance

Unless we manually build a `QueueingDispatcher` instance within the `fake` method.

Reply:
I'm sorry - I can't merge this with this test change. I don't understand this test anymore and we've had too many breaking changes lately.

The test used to plain read that a unique job was dispatched and a lock acquired, then a second job couldn't be dispatched since a lock was in place for the unique job. The test no longer reads that way and is confusing. Please feel free to mark this as ready for review again when the test reads more plainly.

Reply:
It makes more sense if you disregard the "faked" element to this test. This test has been updated to reflect the REAL functionality of the `Bus` facade and underlying manager regardless of the call to to `Bus::fake()`.

In userland, you're not going to call `Bus::fake()`, so if you called `UniqueTestJob::dispatch()` twice in userland, like the test, then the the expectation is that **this job was not dispatched more than once**.

This PR identifies a problem with using `Bus::getFacadeRoot()` and passing the result to `BusFake` – calling `Bus::fake()` twice means you end up with the second`BusFake` instance deferring to the first `BusFake` instance rather than the underlying `Dispatcher` instance (which is what it should do). In any case, calling `Bus::fake()` (or any Facade::fake()`) more than once in a test should be considered bad practice as it leads to incorrect behaviour where `getFacadeRoot` is concerned.

If it helps readability, I can add an `assertDispatchedOnce` method to the `BusFake` class and update the test accordingly.